### PR TITLE
Adding move prover to status checks

### PR DESCRIPTION
**Pull request by ebmifa**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -161,7 +162,7 @@ jobs:
 
       - name: Start local Sui network
         run: |
-          sui start --no-full-node &
+          cargo run --bin sui start --no-full-node &
```
Comment (awelc) [inline .github/workflows/rust.yml:165]: This is unrelated change, right?

**Review by emmazzz — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (ebmifa) [reply .github/workflows/rust.yml:165]: oops, not sure how this got in there. Removing

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -248,3 +249,34 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: EmbarkStudios/cargo-deny-action@v1
+
+  move-prover:
+    name: move-prover
+    needs: diff
+    if: needs.diff.outputs.isMove == 'true'
+    runs-on: [ubuntu-ghcloud]
```
Comment (awelc) [inline .github/workflows/rust.yml:257]: Could we please add a timeout just in case, say 20 minutes?

**Review by awelc — APPROVED** [review]
Other than the timeout request, looks good to me!

File: .github/workflows/rust.yml (hunk 2 above)
Comment (ebmifa) [reply .github/workflows/rust.yml:257]: 20 minutes? Seems like a long time for a required check, right?

Comment (awelc) [reply .github/workflows/rust.yml:257]: Average running time (https://github.com/MystenLabs/sui/actions/workflows/move-prover.yml) is about 7min and the longest I see there is 13min, so I figured 20min timeout would be useful just as fail-safe.

Comment (ebmifa) [reply .github/workflows/rust.yml:257]: Ok, added a 20 minute wait.

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -248,3 +249,38 @@ jobs:
...
+
+    steps:
+      - name: Sleep for 20 minutes
+        run: sleep 1200
```
Comment (awelc) [inline .github/workflows/rust.yml:261]: Does it mean that this check will now always run for 20 minutes? I know very little about this setup but I did see the `timeout-minutes: 45` clause earlier in this file. Is there a difference between the two options?

Comment (ebmifa) [reply .github/workflows/rust.yml:261]: Oh yeah, setting time out is more consistent with the rest of the steps. Let me update.

**Review by awelc — APPROVED** [review]
LGTM!
